### PR TITLE
Add support for getting user id

### PR DESCRIPTION
**Pull request by blagoev**
For #1090

### CODE REVIEW

File: lib/browser/user.js [hunk 1]
```diff
@@ -35,7 +35,8 @@ export default class User {
 
 createMethods(User.prototype, objectTypes.USER, [
     'logout',
-    '_sessionForOnDiskPath'
+    '_sessionForOnDiskPath',
+    'retrieveAccount'
```
Comment (fealebenpae) [inline lib/browser/user.js]: This is only needed for methods implemented in C++.

**Comment by blagoev** [comment]
@fealebenpae  can you please review the last changes.

File: lib/user-methods.js [hunk 2]
```diff
@@ -229,6 +229,33 @@ module.exports = {
...
             });
-        }
-    }
+        },
+        retrieveAccount(provider, provider_id) {
+            checkTypes(arguments, ['string', 'string']);
+
+            let url = url_parse(this.server);
+            url.set('pathname', `/api/providers/${provider}/accounts/${provider_id}`);
+            let headers = {
```
Comment (kristiandupont) [inline lib/user-methods.js]: `const`

File: lib/user-methods.js [hunk 3]
```diff
@@ -229,6 +229,33 @@ module.exports = {
...
+                Authorization: this.token
+            };
+
+            const options = {
+                method: 'GET',
+                headers: headers,
```
Comment (kristiandupont) [inline lib/user-methods.js]: Could just be `headers`

File: tests/js/admin-user-helper.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                        url: `realm://localhost:9080/__admin`,
+                        error: err =>
+                         console.log('Error opening __admin realm ' + err.user  + ' ' + err.url + ' ' + err.state),
+                    }
+                };
+
+                Realm.open(config).then(realm => {
+                    if (userIdentity == admin_token_user.identity)
+                    {
+                        console.log(".");
```
Comment (kristiandupont) [inline tests/js/admin-user-helper.js]: What's the purpose of this?

File: tests/js/admin-user-helper.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                        }
+
+                        Realm.Sync.User.login('http://localhost:9080', newAdminName, password, (error, newAdminUser) => {
+                            if (error) {
+                                reject(error);
+                            } else {
+                                let isAdmin = newAdminUser.isAdmin;
+                                user.logout();
+                                if (!isAdmin) {
+                                    setTimeout(_ => {
```
Comment (kristiandupont) [inline tests/js/admin-user-helper.js]: Wouldn't `() => {...}` be better?

File: tests/js/admin-user-helper.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                                        waitForServerToUpdateAdminUser();
+                                    }, 500);
+                                    return;
+                                }
+
+                                resolve({ 
+                                    username: newAdminName, 
+                                    password: password
```
Comment (kristiandupont) [inline tests/js/admin-user-helper.js]: Could just be `password`

File: tests/js/index.js [hunk 7]
```diff
@@ -41,10 +41,11 @@ if (Realm.Sync) {
     TESTS.SessionTests = require('./session-tests');
 }
 
-function node_require(module) { return require(module); }
+function node_require(module) { return require(module); }
 
 // If on node, run the async tests
-if (typeof process === 'object' && process + '' === '[object process]') {
+global.isNodeProcess = typeof process === 'object' && process + '' === '[object process]'
```
Comment (kristiandupont) [inline tests/js/index.js]: I think this runs in the browser as well, when running tests with chrome debugging? In which case, I guess `global` is undefined.

File: tests/js/admin-user-helper.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+                    }
+                };
+
+                Realm.open(config).then(realm => {
+                    if (userIdentity == admin_token_user.identity)
+                    {
+                        console.log(".");
+                    }
+
+                    let pendingAdminUser = realm.objects('User').filtered(`id == "${userIdentity}"`)[0];
```
Comment (fealebenpae) [inline tests/js/admin-user-helper.js]: Is `id` a primary key? We could potentially use `objectForPrimaryKey` then.

File: tests/js/admin-user-helper.js (hunk 4 above)
Comment (fealebenpae) [inline tests/js/admin-user-helper.js]: What's this?

File: tests/js/index.js [hunk 9]
```diff
@@ -73,6 +74,22 @@ exports.registerTests = function(tests) {
...
 };
 
+exports.prepare = function(done) {
+    if (!global.isNodeProcess || global.testAdminUserInfo) {
+        done();
+    }
+
+    let helper = require('./admin-user-helper');
+    helper.createAdminUser().then(userInfo => {
+        global.testAdminUserInfo = userInfo;
```
Comment (fealebenpae) [inline tests/js/index.js]: What if we have the tests call `createAdminUser()` directly to always get a fresh admin user?

File: tests/js/admin-user-helper.js (hunk 4 above)
Comment (blagoev) [inline tests/js/admin-user-helper.js]: left over. forgot to remove

File: tests/js/admin-user-helper.js (hunk 5 above)
Comment (blagoev) [inline tests/js/admin-user-helper.js]: using it as a shorthand for fast typing. indeed _ variable is not needed

File: tests/js/index.js (hunk 7 above)
Comment (blagoev) [inline tests/js/index.js]: indeed. will refactor

File: tests/js/admin-user-helper.js (hunk 4 above)
Comment (blagoev) [inline tests/js/admin-user-helper.js]: left over + checking anyone pays attention :). removing it right away

File: tests/js/admin-user-helper.js (hunk 8 above)
Comment (blagoev) [inline tests/js/admin-user-helper.js]: just copied cocoa logic. will try with objectForPrimaryKey

File: tests/js/index.js (hunk 9 above)
Comment (blagoev) [inline tests/js/index.js]: I guess the idea is to use only the global admin user. which is always fresh for every test run. I think cocoa reuses this user throughout their tests

Comment (fealebenpae) [inline tests/js/index.js]: I see. Let's leave it as it is and revisit if we need to later on.

File: tests/js/admin-user-helper.js [hunk 10]
```diff
@@ -65,15 +58,15 @@ exports.createAdminUser = function () {
                                 let isAdmin = newAdminUser.isAdmin;
                                 user.logout();
                                 if (!isAdmin) {
-                                    setTimeout(_ => {
+                                    setTimeout(() => {
```
Comment (fealebenpae) [inline tests/js/admin-user-helper.js]: I think this could be just `setTimeout(waitForServerToUpdateAdminUser, 500)`

**Comment by fealebenpae** [comment]
Oh, I totally forgot - could you also please make a new PR with a changelog entry and an update to the API docs?

**Comment by blagoev** [comment]
sure thing. i forgot as well thanks for reminding.

**Comment by blagoev** [comment]
@fealebenpae made this one https://github.com/realm/realm-js/pull/1127

**Pull request merged**